compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -972,7 +975,7 @@ fn check_associated_item(
                     hir_sig.decl,
                     item.def_id.expect_local(),
                 );
-                check_method_receiver(wfcx, hir_sig, item, self_ty);
+                res = res.and(check_method_receiver(wfcx, hir_sig, item, self_ty));
```

Comment:
can we just bail here with a `?`, or does that cause too many other errors to be skipped?

Reply:
Like I'd rather just do `check_method_receiver(..)?` and then return `Ok(())` below

Reply:
I'm doing this dance to avoid situations where you fix an error just to get a new one, but I do not actually have any examples to show (though I could proably construct some). That said, I'm mostly doing all this to have less resistance to my changes 😆 I'd prefer `?` for most sites.

Reply:
I'd be happy if you just do `?`. No resistance from my end 😆